### ISSUE TITLE
[Feature Request] Add Playback Reporting / Jellystat information in Manage Movie / TV Show sidebar

### ISSUE THREAD

**Issue by flosoft** [Fallenbagel/jellyseerr#981]
### Description

One great feature that Overseerr has is a Tautulli integration, showing playback stats in the last day / week / month and the users that have played the content.
This makes it easier to sift through requests that can be deleted as they've been viewed by the requester.

### Desired Behavior

It would be great to have an integration of Jellystat or the Playback Reporting Plugin data that shows on up in the Manage Movie / TV Show sidebar.
It could show the playback count in the last day / week / month / lifetime as separate counts as well as whether the requester has played the content and icons of the others users that have played the content.

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

**Comment by gauthier-th** [comment]
Yes that's a good idea. I'm currently using a custom tool I made to see playback stats per item and who watched what content.

I think it should be better to use the Playback Reporting plugin, as it can be easily installed in the Jellyfin plugins, and it works just fine for this use case.

**Comment by Schrottfresser** [comment]
The Playback Reporting plugin has some nice API endpoints to work with. Although it does only save the history for a configured amount of time (by default 6 Months i think).
I am currently working on implementing the Playback Reporting plugin to advance this Issue. I could save the playback data in the database and make a initial and long periodic complete scan, like Jellyseerr does with the Library too, at every users Items endpoint with the UserData which includes if the Item was played by the specific user and reconstruct the playback history like this. Then i could continue with light calls at the Playback Reporting plugin for shorter periodic scans. There are only three strings to save per playback of a Item.
Does this seem like a valid approach for you all (especially @gauthier-th)?
I would like to have complete playback history to display it in the UI.

**Comment by gauthier-th** [comment]
> Although it does only save the history for a configured amount of time (by default 6 Months i think).

I don't what's the default duration, but I think we should work with what's already configured instead of saving again the data. Instead we can add a warning to say it's better to set "Data Retention" as "Forever" when configuring it in Jellyseerr.

> I could save the playback data in the database

The playback reporting already has a database, it doesn't feel right to duplicate the data from their database into our database.
Why not querying the original database directly instead?

**Comment by Schrottfresser** [comment]
@gauthier-th What do you mean with "original database"? The Jellyfin database or the plugin one?
Showing a warning may be an good idea to not need to call two APIs and I dont like the data duplication either.
What do you think about calling the Jellyfin Items endpoint? This would need a call per User to query through to get full playback data but remove the need for the plugin itself and the warning.

**Comment by Schrottfresser** [comment]
Actually not even the Playback Reporting Plugin does return a list of played Items. So I guess it wont help me in my case. Would it be reasonable to add a optional Jellyseerr plugin to Jellyfin to report global UserData for each Item? We could also extend functionality in future.
Or maybe it isnt even needed to receive the played status and it may be sufficient if a playback started once for what the Playback Reporting Plugin would be suited perfectly.

**Comment by gauthier-th** [comment]
> What do you mean with "original database"? The Jellyfin database or the plugin one?

The plugin one. We could send raw SQL queries to retrieve the data we need.

>  What do you think about calling the Jellyfin Items endpoint? This would need a call per User to query through to get full playback data but remove the need for the plugin itself and the warning.

We could do this, but I'm afraid this will cause a lot of calls regularly when there is a lot of users.

> Actually not even the Playback Reporting Plugin does return a list of played Items.

We can get the list of the played items with a SQL query (in the "query" tab of the Playback Reporting plugin):
<img width="2052" height="821" alt="Image" src="https://github.com/user-attachments/assets/04fb72a8-d607-427a-8246-078b263e3656" />

I suggest we retrieve all the information we need directly by running SQL queries on the plugin database. All the necessary information should be there.

**Comment by Schrottfresser** [comment]
The data stored in the PlaybackActivity table is just about users playing an item and not about that it has been played completely (like the isPlayed flag in Jellyfin UserData). There even is a PlayTime column in that table to indicate that.
If that data is enought for statistics i am fine with it. But maybe the information if an item was fully played (isPlayed) is more interesting and we do not get this from the plugin (at least i do not know of that).

**Comment by gauthier-th** [comment]
> The data stored in the PlaybackActivity table is just about users playing an item and not about that it has been played completely (like the isPlayed flag in Jellyfin UserData). There even is a PlayTime column in that table to indicate that.

Don't you think the PlayTime is reliable enough to determine if a media has been watched? It's not 100% accurate but for 99% of the cases it should be ok?
I don't have anything against fetching the Jellyfin items, I just don't want this feature to hammer the Jellyfin API.

**Comment by Schrottfresser** [comment]
I could assume a item as played if the reported PlayDuration (thats the correct column key) is at least at 90% of the RunTimeTicks (duration of an item). That is basically the same thing that Jellyfin does to determine if a item is played.
The problem with this is that manually marking an item as played will not be recognized by Jellyseerr (although this could also be seen as a positive).

**Comment by gauthier-th** [comment]
> I could assume a item as played if the reported PlayDuration (thats the correct column key) is at least at 90% of the RunTimeTicks (duration of an item). That is basically the same thing that Jellyfin does to determine if a item is played.

Yep, I was thinking about that.

> The problem with this is that manually marking an item as played will not be recognized by Jellyseerr (although this could also be seen as a positive).

I personally see it as a positive thing. For my stats, I prefer to have real data on how much time users spent watching something than to have what users marked as watched (even if they didn't watch it).
But maybe both are not incompatible?

**Comment by CopperGiraffe** [comment]
Indeed would love to see this implemented 👍 If there is any help needed
